nncf/pruning/filter_pruning/algo.py

```diff
@@ -240,8 +243,10 @@ def _find_layerwise_pruning_rate(self, target_flops_pruning_rate):
 
     def set_pruning_rate(self, pruning_rate):
         # Pruning rate from scheduler can be flops pruning rate or percentage of params that should be pruned
+        same_pruning_rate = np.isclose(pruning_rate, self.pruning_rate)
         self.pruning_rate = pruning_rate
-        if not self.frozen:
+        if not self.frozen and not same_pruning_rate:
```

Comment:
I didn't get why do we need this `same_pruning_rate`. To me, here we should be controlled only the desired pruning level and set masks.

Reply:
It is prudent to skip setting one and the same pruning rate to save computation time. However, @mkaglins, now I'm wondering - if the PR schedule sets up a constant PR over many epochs, but the model's weights evolve drastically over these epochs (e.g. because of a high LR setting), should `set_pruning_rate` be called anyway? The algorithm that calculates *which* filters to prune depends on the values of the weights inside the affected filters, does it not?

Reply:
@vshampor You are right, `set_pruning_rate` should be called anyway, I will fix it.
Moreover, deleting `set_pruning_rate` from the scheduler is enough to save computation time.